op-challenger/fault/agent.go

```diff
@@ -65,10 +75,32 @@ func (a *Agent) Act(ctx context.Context) error {
 	return nil
 }
 
+// ShouldResolve returns true if the agent should resolve the game.
+func (a *Agent) ShouldResolve(ctx context.Context, status uint8) bool {
+	expectedStatus := types.GameStatusDefenderWon
+	if a.agreeWithProposedOutput {
+		expectedStatus = types.GameStatusChallengerWon
+	}
+	if expectedStatus != types.GameStatus(status) {
+		a.log.Warn(
```

Comment:
this can be a single line or if it needs to be multline could be a max of 2 lines